components/ai_chat/core/browser/engine/engine_consumer_conversation_api_v2_unittest.cc

```diff
@@ -0,0 +1,309 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/ai_chat/core/browser/engine/engine_consumer_conversation_api_v2.h"
+
+#include <optional>
+#include <string>
+#include <vector>
+
+#include "base/functional/callback_helpers.h"
+#include "base/json/json_writer.h"
+#include "base/memory/scoped_refptr.h"
+#include "base/run_loop.h"
+#include "base/test/bind.h"
+#include "base/test/task_environment.h"
+#include "base/test/test_future.h"
+#include "base/test/values_test_util.h"
+#include "base/values.h"
+#include "brave/components/ai_chat/core/browser/engine/conversation_api_v2_client.h"
+#include "brave/components/ai_chat/core/browser/engine/engine_consumer.h"
+#include "brave/components/ai_chat/core/browser/engine/extended_content_block.h"
+#include "brave/components/ai_chat/core/browser/engine/oai_message_utils.h"
+#include "brave/components/ai_chat/core/browser/model_service.h"
+#include "brave/components/ai_chat/core/common/mojom/ai_chat.mojom.h"
+#include "brave/components/ai_chat/core/common/mojom/common.mojom.h"
+#include "brave/components/ai_chat/core/common/pref_names.h"
+#include "brave/components/ai_chat/core/common/prefs.h"
+#include "components/sync_preferences/testing_pref_service_syncable.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "testing/gmock/include/gmock/gmock.h"
+#include "testing/gtest/include/gtest/gtest.h"
+#include "third_party/abseil-cpp/absl/strings/str_format.h"
+
+using ::testing::_;
+
+namespace ai_chat {
+
+namespace {
+
+constexpr int kTestingMaxAssociatedContentLength = 100;
+
+struct GenerateRewriteTestParam {
+  std::string name;
+  mojom::ActionType action_type;
+  ExtendedContentBlockType expected_content_type;
+  std::string expected_tone;
```

Comment:
https://github.com/brave/brave-core/pull/32663#discussion_r2591072669

Reply:
can we also use expected payload for non change tone message so we know it is from extended content block?
```

    expected_messages = absl::StrFormat(
        R"([
          {
            "role": "user",
            "content": [
              {"type": "brave-page-excerpt", "text": "%s"},
              {"type": "%s", "text": ""}
            ]
          }
        ])",
```

Reply:
https://github.com/brave/brave-core/pull/32663/commits/4efc240dc93082e733a5d464ddf6309942be1795